core/src/test/scala/unit/kafka/log/LogValidatorTest.scala

```diff
@@ -1250,6 +1250,54 @@ class LogValidatorTest {
     testBatchWithoutRecordsNotAllowed(NoCompressionCodec, DefaultCompressionCodec)
   }
 
+  @Test
+  def testInvalidTimestampExceptionHasRelativeOffset(): Unit = {
+    val now = System.currentTimeMillis()
+    val records = createRecords(magicValue = RecordBatch.MAGIC_VALUE_V2, timestamp = now - 1001L,
+      codec = CompressionType.GZIP)
+    try {
+      LogValidator.validateMessagesAndAssignOffsets(
```

Comment:
We should use `assertThrows`. Currently this test will pass if `validateMessagesAndAssignOffsets` returns successfully.

Reply:
I used try - catch in this case because I wanted to test against the exception object if it correctly has `InvalidTimestampException` and `recordErrors`

Reply:
You can use `assertThrows`. It will return the exception which can then be used for assertions.

Reply:
As @hachikuji this can still be done with the `assertThrows` call, you just pass in the statement(s) that you want to cover with the expected throw type, should be easier to use too :)

Reply:
Actually `assertThrows` only returns a `Succeeded` object if the enclosed statement throws an expected exception. I changed it to use `intercept` to do assertion on the exception objects